File: mobile/src/main/java/uk/co/appsbystudio/geoshare/utils/setup/fragments/SetupProfileFragment.java
package uk.co.appsbystudio.geoshare.utils.setup.fragments;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.storage.FileDownloadTask;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;

import java.io.File;

import de.hdodenhof.circleimageview.CircleImageView;
import uk.co.appsbystudio.geoshare.R;

public class SetupProfileFragment extends Fragment {

    public SetupProfileFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        final View view = inflater.inflate(R.layout.fragment_setup_profile, container, false);

        FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
        FirebaseUser user = firebaseAuth.getCurrentUser();

        if (user != null) {
            String userId = user.getUid();

            StorageReference storageReference = FirebaseStorage.getInstance().getReference();

            StorageReference profileRef = storageReference.child("profile_pictures/" + userId + ".png");

            profileRef.getDownloadUrl().addOnSuccessListener(new OnSuccessListener<Uri>() {
                @Override
                public void onSuccess(Uri uri) {
                    TextView descProfile = view.findViewById(R.id.descriptionProfile);
                    Button setPictureButton = view.findViewById(R.id.setPictureButton);

                    descProfile.setText(R.string.desc_picture_is_set);
                    setPictureButton.setText(R.string.upload_new_picture);
                }
            });
        }

        return view;
    }

}

File: mobile/src/main/java/uk/co/appsbystudio/geoshare/setup/fragments/profile/SetupProfileFragment.kt
package uk.co.appsbystudio.geoshare.setup.fragments.profile

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_setup_profile.*
import uk.co.appsbystudio.geoshare.R
import uk.co.appsbystudio.geoshare.setup.InitialSetupActivity
import uk.co.appsbystudio.geoshare.setup.InitialSetupView

class SetupProfileFragment : Fragment(), SetupProfileView {

    private var fragmentCallback: InitialSetupView? = null
    private var presenter: SetupProfilePresenter? = null

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        try {
            fragmentCallback = context as InitialSetupActivity
        } catch (e: ClassCastException) {
            throw ClassCastException(activity.toString() + "must implement InitialSetupView")
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        presenter = SetupProfilePresenterImpl(this)

        return inflater.inflate(R.layout.fragment_setup_profile, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        presenter?.imageAvailable()

        button_set_picture_profile.setOnClickListener {
            fragmentCallback?.onShowProfileDialog()
        }

        button_back_profile.setOnClickListener {
            fragmentCallback?.onBack()
        }

        button_next_profile.setOnClickListener {
            fragmentCallback?.onNext()
        }
    }

    override fun updateUIText() {
        descriptionProfile.setText(R.string.desc_picture_is_set)
        button_set_picture_profile.setText(R.string.upload_new_picture)
    }
}
